simpleast-core/src/test/java/com/discord/simpleast/code/TypeScriptRulesTest.kt

```diff
@@ -0,0 +1,261 @@
+            codeStyleProviders.defaultStyleProvider,
+            CodeRules.createCodeLanguageMap(codeStyleProviders))
+        )
+        .addRules(SimpleMarkdownRules.createSimpleMarkdownRules())
+    treeMatcher = TreeMatcher()
+    treeMatcher.registerDefaultMatchers()
+  }
+
+  @Test
+  fun comments() {
+    val ast = parser.parse("""
+      ```ts
+      /** Multiline
+          Comment
+      */
+      foo.bar(); // Inlined
+      // Line comment
+      
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+          /** Multiline
+              Comment
+          */
+        """.trimIndent(),
+        "// Inlined",
+        "// Line comment")
+  }
+
+  @Test
+  fun strings() {
+    val ast = parser.parse("""
+      ```ts
+      x = 'Hello';
+      y = "world";
+      log(`Hi`);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "'Hello'",
+        "\"world\"",
+        "`Hi`")
+  }
+  @Test
+  fun stringsMultiline() {
+    val ast = parser.parse("""
+      ```ts
+      text = `
+      hello
+      world
+      `;
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+          `
+          hello
+          world
+          `
+        """.trimIndent(),
+    )
+  }
+
+  @Test
+  fun functions() {
+    val ast = parser.parse("""
+      ```ts
+      function test(T) {
+        // Implementation
+      }
+      fn = function () {};
+      function* generator() {}
+      static test() {}
+      async fetch() {}
+      get tokens() {}
+      set constants() {}
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<TypeScript.FunctionNode<*>>(
+      "function test",
+      "function ",
+      "function* generator",
+      "static test",
+      "async fetch",
+      "get tokens",
+      "set constants")
+  }
+
+  @Test
+  fun commentedFunction() {
+    val ast = parser.parse("""
+      ```ts
+      /*
+        function test(T) {
+          throw new Error();
+        }
+      */
+      // function O() {}
+      log(x /* test var */);
+      ```
+    """.trimIndent(), TestState())
+
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        """
+        /*
+          function test(T) {
+            throw new Error();
+          }
+        */
+        """.trimIndent(),
+        "// function O() {}",
+        "/* test var */")
+  }
+
+  @Test
+  fun keywords() {
+    val ast = parser.parse("""
+      ```ts
+      while (true) {}
+      for (;;) {}
+      if (false) {}
+      class {}
+      try {
+      } catch (err) {
+        throw
+      } finally {
+        return;
+      }
+      ```
+    """.trimIndent(), TestState())
+    ast.assertNodeContents<StyleNode.TextStyledNode<*>>(
+        "while", "true", 
+        "for", "if", "false",
+        "class", "try", "catch",
+        "finally", "return", "throw")
```

Comment:
```suggestion
        "throw", "finally", "return")
```
Test fails otherwise because of order.